sources/packages/backend/libs/integrations/src/institution-integration/ier12-integration/ier12-file-detail.ts

```diff
@@ -1,114 +1,578 @@
-        NUMBER_FILLER,
-      );
-      record.appendWithStartFiller(
-        this.exceptionExpenses.toString(),
-        8,
-        NUMBER_FILLER,
-      );
-      record.appendWithStartFiller(
-        this.totalFundedWeeks.toString(),
-        2,
-        NUMBER_FILLER,
-      );
-      record.repeatAppend(SPACE_FILLER, 3); //We have hardcoded the courseLoad to null as its only for FullTime.
-      record.append("F", 1); //This implementation is only for FullTime, so hardcoding it to F.
-      record.appendWithEndFiller(disbursement.coeStatus, 10, SPACE_FILLER);
-      record.appendWithEndFiller(
-        disbursement.disbursementScheduleStatus,
-        10,
-        SPACE_FILLER,
-      );
-      record.appendDate(disbursement.disbursementDate, DATE_FORMAT);
-      disbursement.dateSent
-        ? record.appendDate(disbursement.dateSent, DATE_FORMAT)
-        : record.repeatAppend(SPACE_FILLER, 8);
-      disbursement.disbursementValues.forEach(
-        (disbursementValue: DisbursementValue) => {
-          record.append(disbursementValue.valueCode, 4);
-          record.appendWithStartFiller(
-            round(disbursementValue.valueAmount),
-            8,
-            NUMBER_FILLER,
-          );
-        },
-      );
-      return record.toString();
+    const record = new StringBuilder();
+    record.appendWithStartFiller(this.assessmentId, 10, NUMBER_FILLER);
+    record.appendWithStartFiller(this.disbursementId, 10, NUMBER_FILLER);
+    record.append(this.applicationNumber, 10);
+    record.appendWithEndFiller(
+      emptyStringFiller(this.institutionStudentNumber),
+      12,
+      SPACE_FILLER,
+    );
+    record.append(this.sin, 9);
+    record.appendWithEndFiller(this.studentLastName, 25, SPACE_FILLER);
+    record.appendWithEndFiller(
+      emptyStringFiller(this.studentGivenName),
+      15,
+      SPACE_FILLER,
+    );
+    record.appendDate(this.studentBirthDate, DATE_FORMAT);
+    record.appendWithEndFiller(this.studentGroupCode, 4, SPACE_FILLER);
+    record.appendWithEndFiller(
+      emptyStringFiller(this.studentMaritalStatusCode),
+      4,
+      SPACE_FILLER,
+    );
+    record.appendWithEndFiller(
+      emptyStringFiller(this.studentDisabilityStatusCode),
+      4,
+      SPACE_FILLER,
+    );
+    record.appendWithEndFiller(
+      emptyStringFiller(this.applicationDisabilityStatusFlag),
+      1,
+      SPACE_FILLER,
+    );
+    record.appendWithEndFiller(this.addressInfo.addressLine1, 25, SPACE_FILLER);
+    record.appendWithEndFiller(
+      emptyStringFiller(this.addressInfo.addressLine2),
+      25,
+      SPACE_FILLER,
+    );
+    record.appendWithEndFiller(this.addressInfo.city, 25, SPACE_FILLER);
+    record.appendWithEndFiller(this.addressInfo.provinceState, 4, SPACE_FILLER);
+    record.appendWithEndFiller(this.addressInfo.postalCode, 16, SPACE_FILLER);
+    record.appendWithEndFiller(this.programName, 25, SPACE_FILLER);
+    record.appendWithEndFiller(this.programDescription, 50, SPACE_FILLER);
```

Comment:
Just wondering, our db `programDescription` length is 500, so is the plan to only sent first 50 character, 
![image](https://github.com/bcgov/SIMS/assets/77353155/6f41584f-cd0c-461c-b903-0cebede34c39)

Reply:
Yes. As per the IER12 official document we are using only 50 characters.